src/index.ts

```diff
@@ -1,212 +1,15 @@
-    // Error Reporting will pick up error messages if the full stack trace is
-    // included in the textPayload or the message property of the jsonPayload.
-    // https://cloud.google.com/error-reporting/docs/formatting-error-messages
-    // We prefer to format messages as jsonPayload (by putting it as a message
-    // property on an object) as that works is accepted by Error Reporting in
-    // for more resource types.
-    //
-    if (metadata && metadata.stack) {
-      msg += (msg ? ' ' : '') + (metadata as types.Metadata).stack;
-      data.serviceContext = this.serviceContext;
-    }
-    data.message = this.prefix ? `[${this.prefix}] ` : '';
-    data.message += msg;
-
-    if (is.default.object(metadata)) {
-      data.metadata =
-          this.inspectMetadata ? mapValues(metadata, util.inspect) : metadata;
-
-      // If the metadata contains a httpRequest property, promote it to the
-      // entry metadata. This allows Stackdriver to use request log formatting.
-      // https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest
-      // Note that the httpRequest field must properly validate as HttpRequest
-      // proto message, or the log entry would be rejected by the API. We no do
-      // validation here.
-      if (metadata.httpRequest) {
-        entryMetadata.httpRequest = metadata.httpRequest;
-        delete data.metadata!.httpRequest;
-      }
-
-      // If the metadata contains a labels property, promote it to the entry
-      // metadata.
-      // https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry
-      if (metadata.labels) {
-        entryMetadata.labels = (entryMetadata.labels) ?
-            Object.assign({}, entryMetadata.labels, metadata.labels) :
-            metadata.labels;
-        delete data.metadata!.labels;
-      }
-    }
-
-    if (metadata && metadata[LOGGING_TRACE_KEY]) {
-      entryMetadata.trace = metadata[LOGGING_TRACE_KEY];
-      delete data.metadata![LOGGING_TRACE_KEY];
-    } else {
-      const trace = getCurrentTraceFromAgent();
-      if (trace) {
-        entryMetadata.trace = trace;
-      }
-    }
-
-    const entry = this.stackdriverLog.entry(entryMetadata, data);
-    this.stackdriverLog[stackdriverLevel](entry, callback);
-  }
-}
-// We need to add StackdriverLogging to winston.transport which does not
-// contain this type.
-// tslint:disable-next-line:no-any
-(winston.transports as any).StackdriverLogging = LoggingWinston;
+// this is a class.
+// tslint:disable-next-line:variable-name
+export const LoggingWinston = semver.lt(winstonVersion, '3.0.0') ?
+    require('./winston2').LoggingWinston :
+    require('./winston3').LoggingWinston;
+// winstons are required here so they dont get executed unless they are the
+// correct versions. winston2 inherits from a class provided by winston <3. if
+// its ever just left as an es6 class that symbol will have to be available to
```

Comment:
Thanks for the explanation. What is "it" in this sentence, and what does it mean to "leave it as an es6 class"?

Reply:
Also, newline in next line

Reply:
Does `LoggingWinston` have useful types here (i.e. not `any`)?

Reply:
minor nit: `s/its/it's`.